crates/relayer/src/chain/ckb4ibc.rs

```diff
@@ -624,119 +670,82 @@ impl ChainEndpoint for Ckb4IbcChain {
         &mut self,
         tracked_msgs: TrackedMsgs,
     ) -> Result<Vec<IbcEventWithHeight>, Error> {
-        let mut txs = Vec::new();
-        let mut tx_hashes = Vec::new();
-        let mut events = Vec::new();
         let mut result_events = Vec::new();
-        for msg in tracked_msgs.msgs {
-            let converter = self.get_converter()?;
-            let CkbTxInfo {
-                unsigned_tx,
-                envelope,
-                input_capacity,
-                event,
-            } = convert_msg_to_ckb_tx(msg, &converter)?;
-            if unsigned_tx.is_none() {
-                if let Some(e) = event {
-                    if let IbcEvent::CreateClient(e) = &e {
-                        let client_type = e.0.client_type;
-                        info!("the counterparty client type of Ckb4Ibc is set as {client_type}");
+        let mut msgs = tracked_msgs.msgs;
+        let mut retry_times = 0;
+        let sync_if_create_client = |event: &IbcEvent| -> Option<ClientType> {
+            if let IbcEvent::CreateClient(e) = event {
+                let client_type = e.0.client_type;
+                info!("counterparty client type of Ckb4Ibc is set to {client_type}");
+                return Some(client_type);
+            }
+            None
+        };
+        while !msgs.is_empty() {
```

Comment:
use `for msg in msg {..`?

Reply:
not yet, if msg failed to send, I'll put msg back into `msgs`